2020-05-07 14:55:51.9119 ERROR Invalid model state for the object. Controller: Companies, action: CreateCompany
2020-05-07 14:56:54.3726 ERROR Invalid model state for the object. Controller: Employees, action: CreateEmployeeForCompany
2020-05-07 16:17:31.4309 ERROR Something went wrong: System.NotImplementedException: The method or operation is not implemented.
   at CompanyEmployees.ActionFilters.ValidationFilterAttribute.OnActionExecuted(ActionExecutedContext context) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\ActionFilters\ValidationFilterAttribute.cs:line 39
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:17:53.3594 ERROR Something went wrong: System.NotImplementedException: The method or operation is not implemented.
   at CompanyEmployees.ActionFilters.ValidationFilterAttribute.OnActionExecuted(ActionExecutedContext context) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\ActionFilters\ValidationFilterAttribute.cs:line 39
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:22:54.2879 INFO Unable to find a company for company id:c9d4c053-49b6-410c-bc78-2d54a9991870
2020-05-07 16:24:22.9599 ERROR Invalid model state for the object. Controller: Companies, action: CreateCompany
2020-05-07 16:24:28.2032 ERROR Invalid model state for the object. Controller: Employees, action: CreateEmployeeForCompany
2020-05-07 16:24:32.4933 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:24:40.4608 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:24:53.0592 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:24:59.5295 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:26:02.8456 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:27:00.3963 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991870 doesn't exist in the database.
2020-05-07 16:27:35.5459 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991870 doesn't exist in the database.
2020-05-07 16:27:48.9651 ERROR Invalid model state for the object. Controller: Companies, action: UpdateCompany
2020-05-07 16:28:08.6452 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:28:16.0549 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:28:32.2396 ERROR Something went wrong: System.ArgumentNullException: Value cannot be null. (Parameter 'entity')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
   at Repository.RepositoryBase`1.Delete(T entity) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\RepositoryBase.cs:line 32
   at Repository.CompanyRepository.DeleteCompany(Company company, Boolean trackChanges) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\Repository\CompanyRepository.cs:line 31
   at CompanyEmployees.Controllers.CompaniesController.DeleteCompany(Guid id) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\Controllers\CompaniesController.cs:line 131
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:31:24.8671 INFO company with id: 9b926c36-6782-4e9d-9cec-08d7f0fc11bd doesn't exist in the database.
2020-05-07 16:31:49.9867 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991870 doesn't exist in the database.
2020-05-07 16:33:36.1170 INFO Company with id: c9d4c053-49b6-410c-bc78-2d54a9991870 doesn't exist in the database.
2020-05-07 16:33:50.4189 INFO Employee with id: 80abbca8-664d-4b20-b5de-024705497d4c doesn't exist in the database.
2020-05-07 16:34:48.0018 ERROR Invalid model state for the object. Controller: Employees, action: UpdateEmployeeForCompany
2020-05-07 16:56:07.2481 INFO company with id: a3ee7554-40e7-4eff-b5a5-08d7f299bc4a doesn't exist in the database.
2020-05-07 16:56:22.4541 INFO company with id: 3d490a70-94ce-4d15-9494-5248280c2ce2 doesn't exist in the database.
2020-05-07 16:56:35.2067 ERROR Something went wrong: System.Collections.Generic.KeyNotFoundException: The given key 'id' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at CompanyEmployees.ActionFilters.ValidateCompanyExistsAttribute.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\ActionFilters\ValidateCompanyExistsAttribute.cs:line 23
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 16:57:48.1483 ERROR Something went wrong: System.Collections.Generic.KeyNotFoundException: The given key 'id' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at CompanyEmployees.ActionFilters.ValidateCompanyExistsAttribute.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\ActionFilters\ValidateCompanyExistsAttribute.cs:line 23
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 17:00:15.1840 ERROR Something went wrong: System.Collections.Generic.KeyNotFoundException: The given key 'id' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at CompanyEmployees.ActionFilters.ValidateCompanyExistsAttribute.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in C:\Users\byron\OneDrive\Visual Studio 2019\Projects\CompanyEmployees\CompanyEmployees\ActionFilters\ValidateCompanyExistsAttribute.cs:line 23
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2020-05-07 19:39:59.5734 INFO company with id: 3d490a70-94ce-4d15-9494-5248280c2ce2 doesn't exist in the database.
2020-05-07 19:40:15.1649 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991873 doesn't exist in the database.
2020-05-07 19:40:20.1578 INFO company with id: 86dba8c0-d178-41e7-938c-ed49778fb52a doesn't exist in the database.
2020-05-07 19:40:44.5782 INFO company with id: a3ee7554-40e7-4eff-b5a5-08d7f299bc4a doesn't exist in the database.
2020-05-07 19:48:07.7931 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991870 doesn't exist in the database.
2020-05-07 19:48:14.5193 INFO company with id: c9d4c053-49b6-410c-bc78-2d54a9991872 doesn't exist in the database.
2020-05-07 19:49:10.4512 INFO Unable to find employee(86dba8c0-d178-41e7-938c-ed49778fb52c) in the database
2020-05-07 19:49:55.8112 INFO Employee with id: 86dba8c0-d178-41e7-938c-ed49778fb52c doesn't exist in the database.
2020-05-07 19:50:09.8531 INFO Company with id: c9d4c053-49b6-410c-bc78-2d54a9991872 doesn't exist in the database.
2020-05-07 19:50:27.9329 INFO Company with id: c9d4c053-49b6-410c-bc78-2d54a9991872 doesn't exist in the database.
2020-05-07 19:50:41.2225 INFO Company with id: c9d4c053-49b6-410c-bc78-2d54a9991872 doesn't exist in the database.
2020-05-07 19:52:33.0475 INFO company with id: a3ee7554-40e7-4eff-b5a5-08d7f299bc4a doesn't exist in the database.
2020-05-07 19:55:06.7194 ERROR Invalid model state for the object. Controller: Employees, action: CreateEmployeeForCompany
2020-05-07 19:56:22.1127 INFO Employee with id: 7e9081eb-d219-4e75-9449-08d7f2b82e7d doesn't exist in the database.
2020-05-07 19:57:53.2218 INFO Employee with id: c80f00ee-7e3b-4244-96b8-1e0e17397866 doesn't exist in the database.
2020-05-07 19:58:00.4061 INFO Employee with id: 398b9071-a3d3-4777-944a-08d7f2b82e7d doesn't exist in the database.
2020-05-07 20:03:10.1212 INFO Employee with id: d43a3c47-e4ba-47dc-b13b-08d7f299bc45 doesn't exist in the database.
2020-05-07 20:04:03.9654 INFO Company with id: 398b9071-a3d3-4777-944a-08d7f2b82e7d doesn't exist in the database.
2020-05-07 20:08:21.5109 INFO company with id: 3d490a70-94ce-4d15-9494-5248280c2ce2 doesn't exist in the database.
2020-05-07 20:10:48.2582 ERROR Object sent from client is null. Controller:Employees Action:CreateEmployeeForCompany
2020-05-07 20:10:48.2582 ERROR Invalid model state for the object. Controller: Employees, action: CreateEmployeeForCompany
2020-05-07 20:11:42.4188 ERROR Object sent from client is null. Controller:Employees Action:CreateEmployeeForCompany
2020-05-07 20:11:42.4188 ERROR Invalid model state for the object. Controller: Employees, action: CreateEmployeeForCompany
2020-05-07 20:15:04.2330 INFO company with id: 31511c7a-5f72-4a29-335b-08d7f0f3c611 doesn't exist in the database.
2020-05-07 20:15:35.8481 INFO Employee with id: 9ef2f77b-029f-4057-b226-08d7f2ba4263 doesn't exist in the database.
2020-05-07 20:15:46.1681 INFO Employee with id: 9ef2f77b-029f-4057-b226-08d7f2ba4263 doesn't exist in the database.
